pkg/util/util.go

```diff
@@ -66,21 +66,20 @@ func RandomString(n int) string {
 	return string(result)
 }
 
-// GetControlPlaneMachine returns a control plane machine from input.
-// Deprecated: use GetControlPlaneMachines.
-func GetControlPlaneMachine(machines []*clusterv1.Machine) *clusterv1.Machine {
+// GetControlPlaneMachines returns a slice containing control plane machines.
+func GetControlPlaneMachines(machines []*clusterv1.Machine) (res []*clusterv1.Machine) {
 	for _, machine := range machines {
 		if IsControlPlaneMachine(machine) {
-			return machine
+			res = append(res, machine)
 		}
 	}
-	return nil
+	return
 }
 
-// GetControlPlaneMachines returns a slice containing control plane machines.
-func GetControlPlaneMachines(machines []*clusterv1.Machine) (res []*clusterv1.Machine) {
-	for _, machine := range machines {
-		if IsControlPlaneMachine(machine) {
+// GetControlPlaneMachinesFromList returns a slice containing control plane machines.
+func GetControlPlaneMachinesFromList(machineList clusterv1.MachineList) (res []clusterv1.Machine) {
```

Comment:
Can we have this return `[]*clusterv1.Machine` instead?

Reply:
Sure